controllers/snapshot/snapshot_adapter.go

```diff
@@ -366,6 +369,29 @@ func (a *Adapter) EnsureGlobalCandidateImageUpdated() (controller.OperationResul
 	return controller.ContinueProcessing()
 }
 
+func GetStatusCondition(snapshot *applicationapiv1alpha1.Snapshot) (condition *metav1.Condition, ok bool) {
+	const (
+		AppStudioTestSucceededCondition = "AppStudioTestSucceeded"
+		LegacyTestSucceededCondition    = "LegacyTestSucceeded"
+	)
+
+	condition = meta.FindStatusCondition(snapshot.Status.Conditions, AppStudioTestSucceededCondition)
+	if condition == nil {
+		condition = meta.FindStatusCondition(snapshot.Status.Conditions, LegacyTestSucceededCondition)
+	}
+
+	ok = (condition != nil && condition.Status != metav1.ConditionUnknown)
+	return
+}
+
+func GetAppStudioTestsFinishedTime(snapshot *applicationapiv1alpha1.Snapshot) (metav1.Time, bool) {
```

Comment:
Same as above, this function might make more sense being placed in [gitops/snapshot.go](https://github.com/redhat-appstudio/integration-service/blob/main/gitops/snapshot.go).